fedimint-client/src/sm/executor.rs

```diff
@@ -416,6 +376,25 @@ where
             })
             .collect::<Vec<_>>();
         if transitions.is_empty() {
+            // In certain cases a terminal (no transitions) state could get here due to
+            // module bug. Inactivate it to prevent accumulation of such states.
+            // See [`Self::add_state_machines_dbtx`].
+            warn!(module_id = module_instance, "A terminal state where only active states are expected. Please report this bug upstream.");
+            self.db
+                .autocommit::<_, _, anyhow::Error>(
+                    |dbtx, _| {
+                        Box::pin(async {
+                            let k = InactiveStateKey::from_state(state.clone());
+                            let v = ActiveState::new().into_inactive();
+                            dbtx.insert_entry(&k, &v).await;
+                            Ok(())
+                        })
```

Comment:
Don't you also have to remove the active state?

Reply:
Oh, right. I copied it from the place that was rejecting the new SM, which didn't have to do that. :facepalm: